src/rgw/rgw_op.cc

```diff
@@ -2129,6 +2129,18 @@ void RGWGetObj::execute(optional_yield y)
       filter = &*decompress;
   }
 
+  attr_iter = attrs.find(RGW_ATTR_MANIFEST);
+  if (attr_iter != attrs.end() && get_type() == RGW_OP_GET_OBJ && get_data) {
+    RGWObjManifest m;
+    decode(m, attr_iter->second);
+    if (m.get_tier_type() == "cloud") {
+      op_ret = -ERR_INVALID_OBJECT_STATE;
```

Comment:
here you could in theory add the redirect code (if configured for such)

Reply:
thanks! I have added it as TODO comment for now.